packages/core/src/helpers/logger/device-console.logger.ts

```diff
@@ -1,42 +1,54 @@
-import { AbstractLogger, LogLevel, UnknownLogLevelError } from './logger';
+import { Logger, LogLevel, UnknownLogLevelError } from './logger';
 
-export class DeviceConsoleLogger extends AbstractLogger {
-    protected logger: Console;
+export class DeviceConsoleLogger extends Logger {
+    private logger: Console;
+    private tag?: string;
 
-    constructor(logger?: Console) {
+    constructor(logger?: Console, tag?: string) {
         super();
-        this.logger = logger || console;
+        this.logger = logger ?? console;
+        this.tag = tag;
     }
 
-    logKeyValue(key: string, value: unknown): void {
+    public logKeyValue(key: string, value: unknown): void {
         this.info(`${key}=${value}`);
     }
 
-    log(level: LogLevel, message: string): void;
-    log(level: LogLevel, tag: string, message: string): void;
-    log(level: LogLevel, tagOrMessage: string, message?: string): void {
-        if (message) {
-            message = `[${tagOrMessage}] ${message}`;
-        } else {
-            message = tagOrMessage;
+    protected createLoggerWithTag(tag: string): Logger {
+        return new DeviceConsoleLogger(this.logger, tag);
+    }
+
+    protected handleLog(level: LogLevel, parameters: unknown[]): void {
+        parameters = parameters ?? [];
```

Comment:
why not having a default value for parameters? It's not a good practice to override an input

Reply:
I've changed it as you suggest, probably more idiomatic. Btw, that doesn't override it, just adds a default in case we get `null/undefined`… e.g. same as putting `= []` in the signature. I suspect you didn't want to say `override`. :P

Reply:
You're right, it's not an override. I suggested it because this can might be detected as an issue when Linting the project.

Reply:
Nope, `null` is treated as a proper value: https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMOcAUBDATlgXI8AazDgHcwBtAXUQF5FqBKRAbwFgAoRRCBAZzgAbAKYA6QXADmmHIwDcnAL6dOKdGBCDBiDHx1gAnvNWo04ACbDgMMMPM69GQ8Y5A

This shouldn't be an issue, as `handleLog` is called via `Logger`, which will always pass an array as it uses the "rest" operator `...`. Actually we can remove the default.